extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/HttpSecurityProcessor.java

```diff
@@ -127,13 +134,46 @@ void setMtlsCertificateRoleProperties(
         }
     }
 
+    @BuildStep(onlyIf = IsApplicationBasicAuthRequired.class)
+    void detectBasicAuthImplicitlyRequired(HttpBuildTimeConfig buildTimeConfig,
+            BeanRegistrationPhaseBuildItem beanRegistrationPhaseBuildItem, ApplicationIndexBuildItem applicationIndexBuildItem,
+            BuildProducer<SystemPropertyBuildItem> systemPropertyProducer,
+            List<EagerSecurityInterceptorBindingBuildItem> eagerSecurityInterceptorBindings) {
+        if (makeBasicAuthMechDefaultBean(buildTimeConfig)) {
+            var appIndex = applicationIndexBuildItem.getIndex();
+            boolean noCustomAuthMechanismsDetected = beanRegistrationPhaseBuildItem
+                    .getContext()
+                    .beans()
+                    .filter(b -> b.hasType(AUTH_MECHANISM_NAME))
+                    .filter(BeanInfo::isClassBean)
+                    .filter(b -> appIndex.getClassByName(b.getBeanClass()) != null)
+                    .isEmpty();
+            // we can't decide whether custom mechanisms support basic auth or not
+            if (noCustomAuthMechanismsDetected) {
+                systemPropertyProducer
```

Comment:
@michalvavrik Oh I see, so you keep it so that when custom Basic mechanisms are used, users can set this property ? I'd just drop it, this is a theoretically possible case but no more at this stage IMHO, I've never heard anyone attempt to customize the basic authentication

Reply:
Or what else this option for ?

Reply:
> Oh I see, so you keep it so that when custom Basic mechanisms are used, users can set this property ?
> Or what else is this option for ?

no no no, we are much didn't understand each other. during the build time I detect that we produce BasicAuthenticationMechanism and there is no custom authentication mechanism. This information needs to be propagated to the runtime to the HTTPAuthenticator so I used the system property. normally if you want to propagate build time information you use synthetic beans, but doing anything with HTTPAuthenticator would be sensitive now.

So to conclude this: this is Quarkus telling Quarkus what it detected during the build time. I think it's acceptable for such the edge case considering alternative is bigger refactoring.

> I'd just drop it, this is a theoretically possible case but no more at this stage IMHO, I've never heard anyone attempt to customize the basic authentication

- user that opened this issue #39912 did it https://github.com/Serkan80/quarkus-issue-reproducers/blob/1d7e988835f9e7772188b05bb237c4f1764c459a/http-path-permission-issue/src/main/java/com/acme/gateway/AuthenticationSelector.java
- I certainly do it in some of the tests
- most importantly, I simply don't know if someone doesn't delegate

if you are confident that users do not delegate to the basic auth, I will drop it. just be warned as there is slight potential for the breaking change @sberyozkin , please confirm

Reply:
> LGTM, the only minor suggestion is to drop TEST_IF_BASIC_AUTH_IMPLICITLY_REQUIRED

I can create alternative mechanism to propagate the information, but if what you really mean is that you want to always check whether the basic auth is there, then it won't bring any information, there are just 2 states:

- bean is there and programmatic lookup works -> that is already covered, we have Basic auth mechanism listed as a bean, we try to add it and warn if there is no identity provider
- bean is there and programmatic lookup excluded if for it is a default bean and there are other mechanisms -> it is covered by this PR and this condition.

Reply:
@michalvavrik Using the system property build item to propagate this info has been confusing